api/v1beta1/azuremachinetemplate_webhook_test.go

```diff
@@ -332,7 +332,6 @@ func TestAzureMachineTemplate_ValidateUpdate(t *testing.T) {
 	for _, amt := range tests {
 		amt := amt
 		t.Run(amt.name, func(t *testing.T) {
-			t.Parallel()
```

Comment:
This was my (bad) advice in #2641. There are two loops which reuse the same test data, so `t.Parallel()` in the first block meant that the `dry-run=false` cases could run simultaneously with the `dry-run=true` cases.